pages/docs/concepts/asyncapi-document/resuable-parts.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Reusable parts
+weight: 280
+---
+
+Reusable parts in AsyncAPI provide flexibility, modularity, and code reusability. Using reusable parts makes it easier to generate code and validate the specifications. You can reuse specific document sections such as Messages or schema definitions.
+
+Reusable parts allow you to split up the AsyncAPI document into many files and reference them using the Reference Object. You can use the $ref keyword to reference the same or another local file or external URL.
+
+## Same file
+
+You can use the $ref keyword to reference a component within the same document. The diagram below defines referencing a component within the same document.
+
+```mermaid
+graph LR
+style A fill:#47BCEE, stroke:#333, stroke-width:2px;
+style C fill:#47BCEE, stroke:#333, stroke-width:2px;
+style D fill:#47BCEE, stroke:#333, stroke-width:2px;
+style B fill:#47BCEE, stroke:#333, stroke-width:2px;
+
+A[MyMessageSchema]
+B[components]
+C[myChannel]
+D[$ref keyword]
+
+C -->|publishes| D
+D -->|references| B
+B -->|references| A
```

Comment:
I don't fully understand the meaning of this diagram. What are you trying to explain here?